ext/opcache/jit/zend_jit_arm64.dasc

```diff
@@ -11839,6 +11838,21 @@ static int zend_jit_fetch_obj(dasm_State          **Dst,
 			|	cbnz FCARG2x, >1
 			|.cold_code
 			|1:
+			|	ldr TMP1w, [FCARG2x, #offsetof(zend_property_info, flags)]
```

Comment:
Second: Is this code still correct when REG2w is pointing to the type info IS_REFERENCE after your refactoring? How should this handle IS_REFERENCE pointing to IS_OBJECT?

The label `1:` is in the branch where there is no property type info `if (!prop_info)`. But below, in the branch where there is non-null type info, I see a conditional jump to the label 1. (`IF_TYPE TMP1w, IS_REFERENCE, >1`)

But it doesn't seem as if FCARG2x has been initialized yet when it's jumping to the label 1 (`IF_TYPE TMP1w, IS_REFERENCE, >1`)? So would this be a conditional branch based on an uninitialized value of FCARG2x?

```c
			} else if (flags == ZEND_FETCH_REF) {
				|	GET_LOW_8BITS TMP1w, REG2w
				|	IF_TYPE TMP1w, IS_REFERENCE, >1
				if (ce && ce->ce_flags & ZEND_ACC_IMMUTABLE) {
					|	LOAD_ADDR FCARG2x, prop_info
				} else {

```

I think that may be why Zend/tests/type_declarations/typed_properties_065.phpt is throwing after calling  `    function &offsetGet($o) { return $this->foo; }` on arm? (if I read that right)

```
class@anonymous::offsetGet: ; (lines=3, args=1, vars=1, tmps=1)
    ; (after optimizer)
    ; /path/to/test.php:7-7
L0 (7):     CV0($o) = RECV 1
L1 (7):     V1 = FETCH_OBJ_W (ref) THIS string("foo")
L2 (7):     RETURN_BY_REF V1

```

(Sorry, trying to learn how the arm jit port works)

Reply:
Note that there are multiple `1:` labels in this function. `>1` means that it will go to the next `1:` label after this code, which would be the later at the end of the ZEND_FETCH_REF branch in this case. `<1` would jump to the previous `1:` label instead. As such, I don't think this is the problem...